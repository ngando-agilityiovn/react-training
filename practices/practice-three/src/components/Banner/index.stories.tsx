import { Meta, StoryObj } from '@storybook/react'
import { BrowserRouter } from 'react-router-dom'

// Components
import Banner from '.'

export default {
  title: 'Components/Banner',
  component: Banner,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
} as Meta

type Story = StoryObj<typeof Banner>

export const Default: Story = {
  args: {},
}
