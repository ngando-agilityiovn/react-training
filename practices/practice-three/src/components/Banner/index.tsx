import { Box, Button, Container, Flex, Heading, Img } from '@chakra-ui/react'

const Banner = () => {
  return (
    <Box
      h="287px"
      background="linear-gradient(75deg, #f4e8f3 0%, #f3eff6 52.07%, #eee0f9 102.02%)"
      pos="relative"
      mx="-80px"
    >
      <Container h="100%" maxW="1280px" p="0">
        <Flex h="100%" justifyContent="space-between">
          <Box pl="55" pt="64px">
            <Heading
              w="377px"
              color="primary"
              fontWeight="bold"
              fontSize="34px"
            >
              Grab Upto 50% Off On Selected Headphone
            </Heading>
            <Button variant="solid" mt="43px" disabled>
              Buy Now
            </Button>
          </Box>
          <Img
            src="https://firebasestorage.googleapis.com/v0/b/card-f9463.appspot.com/o/banner.png?alt=media&token=74d4d6cd-551b-4a73-96b7-5dfed2daf5ae"
            alt="Image of banner"
            pr="98px"
          />
        </Flex>
      </Container>
    </Box>
  )
}

export default Banner
