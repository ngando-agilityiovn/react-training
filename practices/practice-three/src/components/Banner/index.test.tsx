import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

// Components
import Banner from '.'
describe('Banner component', () => {
  it('Render correcty', () => {
    const container = render(
      <BrowserRouter>
        <Banner />
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
