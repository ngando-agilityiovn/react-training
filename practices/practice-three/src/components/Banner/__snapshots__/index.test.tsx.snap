// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component Render correcty 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-jueiek"
      >
        <div
          class="chakra-container css-bpa4bz"
        >
          <div
            class="css-i96wi2"
          >
            <div
              class="css-10ly6rs"
            >
              <h2
                class="chakra-heading css-af3y3i"
              >
                Grab Upto 50% Off On Selected Headphone
              </h2>
              <button
                class="chakra-button css-99vstr"
                type="button"
              >
                Buy Now
              </button>
            </div>
            <img
              alt="Image of banner"
              class="chakra-image css-53qf4y"
              src="https://firebasestorage.googleapis.com/v0/b/card-f9463.appspot.com/o/banner.png?alt=media&token=74d4d6cd-551b-4a73-96b7-5dfed2daf5ae"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-jueiek"
    >
      <div
        class="chakra-container css-bpa4bz"
      >
        <div
          class="css-i96wi2"
        >
          <div
            class="css-10ly6rs"
          >
            <h2
              class="chakra-heading css-af3y3i"
            >
              Grab Upto 50% Off On Selected Headphone
            </h2>
            <button
              class="chakra-button css-99vstr"
              type="button"
            >
              Buy Now
            </button>
          </div>
          <img
            alt="Image of banner"
            class="chakra-image css-53qf4y"
            src="https://firebasestorage.googleapis.com/v0/b/card-f9463.appspot.com/o/banner.png?alt=media&token=74d4d6cd-551b-4a73-96b7-5dfed2daf5ae"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
